client/components/Profiles/UploadMedia.tsx

```diff
@@ -0,0 +1,84 @@
+import { View, Text, Pressable, Alert, ActivityIndicator } from 'react-native'
+import * as ImagePicker from 'expo-image-picker'
+import { firebaseConfig } from '../../firebaseConfig'
+import React, { useState } from 'react'
+import * as FileSystem from 'expo-file-system'
+import { initializeApp } from 'firebase/app'
+import { getStorage, ref, uploadBytes, getDownloadURL } from 'firebase/storage'
+
+const app = initializeApp(firebaseConfig)
+const storage = getStorage(app)
+
+interface UploadMediaFileProps {
+    onImageUpload: (imageUrl: string) => void
+}
+
+const UploadMediaFile = ({ onImageUpload }: UploadMediaFileProps) => {
+    const [uploading, setUploading] = useState<boolean>(false)
+
+
+    const pickAndUploadImage = async () => {
+        const { status } = await ImagePicker.requestMediaLibraryPermissionsAsync()
+
+        if (status !== "granted") {
+            Alert.alert(
+                "Permission Denied",
+                "Sorry, we need camera roll permission to upload images."
+            )
+            return
+        }
+
+        let result = await ImagePicker.launchImageLibraryAsync({
+            mediaTypes: ImagePicker.MediaTypeOptions.Images,
+            allowsEditing: true,
+            aspect: [1, 1],
+            quality: 1,
+        })
+
+        if (result.canceled || !result.assets || !result.assets.length) {
+            return
+        }
+
+        const image = result.assets[0].uri
+        setUploading(true)
+
+        try {
+            const fileInfo = await FileSystem.getInfoAsync(image)
+            const blob = await new Promise<Blob>((resolve, reject) => {
+                const xhr = new XMLHttpRequest()
+                xhr.onload = () => {
+                    resolve(xhr.response)
+                }
+                xhr.onerror = (e) => {
+                    reject(new TypeError('Network request failed'))
+                }
+                xhr.responseType = 'blob'
+                xhr.open('GET', fileInfo.uri, true)
+                xhr.send(null)
+            })
+
+            const filename = image.substring(image.lastIndexOf('/') + 1)
```

Comment:
can you explain the naming convention you are using for the files

Reply:
Saw it on various stack overflow files for uploading images to firebase. Ensures we don't get double ups, etc